esa/esa_test.go

```diff
@@ -0,0 +1,363 @@
+package esa
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	token = "b81f8961eb0a8c53faf69faf71b7d62dab9e581f7d2f5344f7f0d686e9e711f8"
+)
+
+var (
+	// mux is the HTTP request multiplexer used with the test server.
+	mux *http.ServeMux
+
+	// client is the esa client being tested.
+	client *Client
+
+	// server is a test HTTP server used to provide mock API responses.
+	server *httptest.Server
+)
+
+// setup sets up a test HTTP server along with a esa.Client that is
+// configured to talk to that test server. Tests should register handlers on
+// mux which provide mock responses for the API method being tested.
+func setup() {
+	// test server
+	mux = http.NewServeMux()
+	server = httptest.NewServer(mux)
+
+	// github client configured to use test server
+	client = NewClient(nil)
+	url, _ := url.Parse(server.URL)
+	client.BaseURL = url
+}
+
+// teardown closes the test HTTP server.
+func teardown() {
+	server.Close()
+}
+
+//func TestExec(t *testing.T) {
+//ctx := context.Background()
+//ts := oauth2.StaticTokenSource(
+//&oauth2.Token{AccessToken: token},
+//)
+//tc := oauth2.NewClient(ctx, ts)
+//c := NewClient(tc)
+//req, err := c.NewRequest("GET", "/teams", nil)
+//if err != nil {
+//t.Error(err)
+//}
+//v := new(bytes.Buffer)
+//_, err = c.Do(ctx, req, v)
+//if err != nil {
+//t.Error(err)
+//}
+//t.Errorf("%s", v)
+//}
+
+func testURLParseError(t *testing.T, err error) {
```

Comment:
**[unused]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
func testURLParseError is unused (U1000)